app/models/incomeSourceDetails/TaxYear.scala

```diff
@@ -88,6 +98,14 @@ object TaxYear {
     }
   }
 
+  def getTaxYear(localDate: LocalDate): TaxYear = {
```

Comment:
ideally as implicit, so we can convert LocalDate to TaxYear, but this approach is fine as well;